packages/block-editor/src/components/block-variation-picker/index.native.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import {
+	ScrollView,
+	View,
+	Text,
+	TouchableWithoutFeedback,
+	Platform,
+} from 'react-native';
+import { map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect, useDispatch } from '@wordpress/data';
+import { compose, usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { createBlock } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+import {
+	PanelBody,
+	BottomSheet,
+	UnsupportedFooterControl,
+} from '@wordpress/components';
+import { Icon, close } from '@wordpress/icons';
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import MenuItem from '../inserter/menu-item';
+import styles from './style.scss';
+
+const hitSlop = { top: 22, bottom: 22, left: 22, right: 22 };
+
+function createBlocksFromInnerBlocksTemplate( innerBlocksTemplate ) {
+	return map(
+		innerBlocksTemplate,
+		( [ name, attributes, innerBlocks = [] ] ) =>
+			createBlock(
+				name,
+				attributes,
+				createBlocksFromInnerBlocksTemplate( innerBlocks )
+			)
+	);
+}
+
+function BlockVariationPicker( { isVisible, onClose, clientId, variations } ) {
+	const { replaceInnerBlocks } = useDispatch( 'core/block-editor' );
+	const isIOS = Platform.OS === 'ios';
+
+	const cancelButtonStyle = usePreferredColorSchemeStyle(
+		styles.cancelButton,
+		styles.cancelButtonDark
+	);
+
+	const leftButton = useMemo(
+		() => (
+			<TouchableWithoutFeedback onPress={ onClose } hitSlop={ hitSlop }>
+				<View>
+					{ isIOS ? (
+						<Text
+							style={ cancelButtonStyle }
+							maxFontSizeMultiplier={ 2 }
+						>
+							{ __( 'Cancel' ) }
+						</Text>
+					) : (
+						<Icon
+							icon={ close }
+							size={ 24 }
+							style={ styles.closeIcon }
+						/>
+					) }
+				</View>
+			</TouchableWithoutFeedback>
+		),
+		[ onClose, cancelButtonStyle ]
+	);
+
+	const onVariationSelect = ( variation ) => {
+		replaceInnerBlocks(
+			clientId,
+			createBlocksFromInnerBlocksTemplate( variation.innerBlocks ),
+			false
+		);
+		onClose();
+	};
+
+	return (
+		<BottomSheet
+			isVisible={ isVisible }
+			onClose={ onClose }
+			title={ __( 'Select a layout' ) }
+			contentStyle={ styles.contentStyle }
+			leftButton={ leftButton }
+		>
+			<ScrollView
+				horizontal
+				showsHorizontalScrollIndicator={ false }
+				contentContainerStyle={ styles.contentContainerStyle }
+				style={ styles.containerStyle }
+			>
+				{ variations.map( ( v ) => {
+					return (
+						<MenuItem
+							item={ v }
+							key={ v.name }
+							onSelect={ () => onVariationSelect( v ) }
+						/>
+					);
+				} ) }
+			</ScrollView>
+			<PanelBody>
+				<UnsupportedFooterControl
```

Comment:
Could we do something with that we use the `UnsupportedFooterControl` for showing the information related to that block? I thought it should be used only for Unsupported features. Maybe it is worth to rename it somehow?

Reply:
I agree that name is not descriptive any more and it should be rather named `FooterMessageControl`, however, extracting it to the separate component has less sense for me since:

1. it's a temporary message since we don't display columns proportionally yet,
2. the same code is used in 3 places, but there is no common place for a separated component.

Reply:
https://github.com/WordPress/gutenberg/pull/24399